pkg/storage/chunk/cache/memcached.go

```diff
@@ -101,7 +106,12 @@ func NewMemcached(cfg MemcachedConfig, client MemcachedClient, name string, reg
 					batchID: input.batchID,
 				}
 				res.found, res.bufs, res.missed, res.err = c.fetch(input.ctx, input.keys)
-				input.resultCh <- res
+				// NOTE: This check is needed because goutines submitting work via `inputCh` may exit in-between because of context cancellation or timeout. This helps to close these worker goroutines to exit without hanging around.
```

Comment:
```suggestion
				// NOTE: This check is needed because goroutines submitting work via `inputCh` may exit in-between because of context cancellation or timeout. This helps to close these worker goroutines to exit without hanging around.
```

Reply:
Done.